openjdk/src/test/java/org/conscrypt/PlatformTest.java

```diff
@@ -174,10 +177,22 @@ public void test_setAndGetSSLParameters() throws Exception {
                 paramsOut.getEndpointIdentificationAlgorithm());
         assertEquals(paramsIn.getWantClientAuth(), paramsOut.getWantClientAuth());
         assertEquals(paramsIn.getNeedClientAuth(), paramsOut.getNeedClientAuth());
-        assertEquals(paramsIn.getSNIMatchers(), paramsOut.getSNIMatchers());
+        assertSNIMatchersEqual(paramsIn.getSNIMatchers(), paramsOut.getSNIMatchers());
         assertEquals(paramsIn.getAlgorithmConstraints(), paramsOut.getAlgorithmConstraints());
     }
 
+    private static void assertSNIMatchersEqual(Collection<SNIMatcher> a, Collection<SNIMatcher> b) {
+        assertEquals(a.size(), b.size());
+
+        HashSet<SNIMatcher> aSet = new HashSet<>(a);
+        aSet.removeAll(b);
+        assertEquals(0, aSet.size());
+
+        HashSet<SNIMatcher> bSet = new HashSet<>(b);
```

Comment:
if a.size() == b.size() then  you only need one removeAll() check

Reply:
Absolutely right. I'll remove the second one.